### ISSUE TITLE
[Feature] Bypass Proxy Settings for these domains

### ISSUE THREAD

**Issue by Vrutin** [ProxymanApp/Proxyman#283]
[Feature] Bypass Proxy Settings for these domains

Maybe worth adding a box in the "External Proxy Settings" where we can list domains that will be ignored.

### Screenshots (something like this)
<img width="570" alt="Bypass" src="https://user-images.githubusercontent.com/3211220/65127690-e0984f00-d9ef-11e9-9fd2-58df334e8fef.png">

**Comment by NghiaTranUIT** [comment]
That's good point @Vrutin  👍 Plan to ship in Milestone 1.8.0 🌮

**Comment by Vrutin** [comment]
@NghiaTranUIT does ProxyMan currently inherit all the bypass proxy settings from the system?

**Comment by NghiaTranUIT** [comment]
Yes, it's @Vrutin. Proxyman inherits the bypass HTTP Proxy setting (Screenshot) in the app but not for External Proxy, which requires some work on it.

<img width="780" alt="Screen Shot 2019-09-25 at 20 16 46" src="https://user-images.githubusercontent.com/5878421/65604301-6bb9ac00-dfd1-11e9-998d-be2e6ac1e77a.png">

**Comment by crankygeek** [comment]
@NghiaTranUIT Should this inheritance also apply to remote devices? I'm trying to filter out, e.g. *.apple.com, but the traffic still appears in Proxyman.

**Comment by NghiaTranUIT** [comment]
Unfortunately, this by-pass proxy setting doesn't affect traffic from the remote device since it's different 😕 and we couldn't set by-pass domains on iPhone too.

To workaround, you can add *.apple.com as a blacklist.

I will implement this feature in the 2.4.0 version 👍

**Comment by crankygeek** [comment]
I've thought of using the Block List, but I don't want to drop those calls, I just don't want to see them in flow list. I've also thought of using Allow List, which sort of works, but I have to remember to add all the different domains and inevitably I miss some.

What do you think about something like a "Hide" or "Ignore" list? Still allow the requests to go through, but they won't be shown in the flow list?

**Comment by NghiaTranUIT** [comment]
@crankygeek @Vrutin Sorry for the delay, but this feature is finally implemented 😄 

Please check out this beta build: https://proxyman.s3.us-east-2.amazonaws.com/beta/Proxyman_2.27.0_Bypass_External_Proxy.dmg

**Comment by crankygeek** [comment]
@NghiaTranUIT Just confirming that this change still only affects Mac traffic and not remote devices? e.g., if I add `*.yahoo.com` to *Tools > Proxy Settings > External Proxy Settings...*, only traffic from the Mac (or simulators) to `www.yahoo.com` will be ignored. But visiting the same site from an iPhone would log the traffic?

**Comment by crankygeek** [comment]
@NghiaTranUIT Just spotted a bug using the beta build of Version 2.27.0 (22700). While the modal for *External Proxy Settings...* is displayed, if you click the menu option again, when you close the modal, it will reopen. This appears to happen for each time you click the menu option, so if you open it, then click it twice, the modal will reopen and have to be closed two more times.

I wonder if most menu items should be blocked when the modal is open?

**Comment by NghiaTranUIT** [comment]
> @NghiaTranUIT Just spotted a bug using the beta build of Version 2.27.0 (22700). While the modal for _External Proxy Settings..._ is displayed, if you click the menu option again, when you close the modal, it will reopen. This appears to happen for each time you click the menu option, so if you open it, then click it twice, the modal will reopen and have to be closed two more times.
> 
> I wonder if most menu items should be blocked when the modal is open?

Thanks. It's a bug 😄 


> @NghiaTranUIT Just confirming that this change still only affects Mac traffic and not remote devices? e.g., if I add `*.yahoo.com` to _Tools > Proxy Settings > External Proxy Settings..._, only traffic from the Mac (or simulators) to `www.yahoo.com` will be ignored. But visiting the same site from an iPhone would log the traffic?

I've tested and the traffic from iPhone doesn't appear on the External Proxy Server.

Here is how I setup:
1. Open Charles Proxy at port 8080
2. Open Proxyman at port 9090
3. Set External Proxy to 127.0.0.1 port 8080 (Charles Proxy acts as an External Proxy)
4. Set *.producthunt.com in the bypass External Proxy

=> When making a normal request (e.g. google.com), I can see the google traffic appears in both Proxyman app and Charles Proxy app (Because Proxyman proxies to the external proxy, which is Charles Proxy)
=> When making www.producthunt.com from my Macbook, I can see this traffic on the Proxyman app, but there is no traffic on Charles Proxy => It means, it's bypassed the External Proxy.
=> Try again on iPhone, the same result. There is no www.producthunt.com on Charles Proxy => Expected Behavior.

@crankygeek Maybe you misunderstand the Bypass External Proxy. It's not a bypass Proxy 🤔

**Comment by crankygeek** [comment]
Yup, completely misunderstood the feature. 😄
